File: log/src/main/java/com/lookforlog/protocol/constants/RespondStatus.java
package com.lookforlog.protocol.constants;

import com.lookforlog.log.constant.CodedConstant;
import io.netty.handler.codec.http.HttpResponseStatus;

public enum RespondStatus implements CodedConstant {
    SUCCESS(0),
    UNAUTHORIZED(HttpResponseStatus.UNAUTHORIZED.code()),//401
    DECODE_ERROR(HttpResponseStatus.BAD_REQUEST.code()),//400
    INTERNAL_SERVER_ERROR(HttpResponseStatus.INTERNAL_SERVER_ERROR.code()),//500
    ;

    final int code;

    RespondStatus(int code) {
        this.code = code;
    }

    @Override
    public int getCode() {
        return this.code;
    }
}

File: log/ui/src/constant/RespondStatus.js
import PropTypes from 'prop-types';

const RespondStatus = {
  SUCCESS: {
    code: 0,
  },
  UNAUTHORIZED: {
    code: 401,
  },
  DECODE_ERROR: {
    code: 400,
  },
  INTERNAL_SERVER_ERROR: {
    code: 500,
  }
}

export const valueOf = code => Object.values(RespondStatus).find(t => t.code === code) || RespondStatus.SUCCESS;

valueOf.propTypes = {
  code: PropTypes.number.isRequired
}

export default RespondStatus;